### PR TITLE
Add format flag to network and volume ls

### PR DESCRIPTION
**Pull request by vdemeester**
This adds a `--format` flag to the `docker network ls` and `docker volume ls` commands, that
allows to customize the format of the output 🐹.

This also re-organize the `api/client/formatter` package by splitting it into several files (`container.go`, `image.go`, …).

This needs :
- [x] More unit tests on the `api/client/formatter` package
- [x] Integration tests
- [x] Documentation updates

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like there's a failure;

```
15:15:50 ----------------------------------------------------------------------
15:15:50 FAIL: docker_cli_run_test.go:4292: DockerSuite.TestRunNamedVolumeNotRemoved
15:15:50 
15:15:50 docker_cli_run_test.go:4300:
15:15:50     c.Assert(strings.TrimSpace(out), checker.Equals, "test")
15:15:50 ... obtained string = "Template parsing error: template: :1:2: executing \"\" at <.ID>: ID is not a field of struct type *formatter.volumeContext"
15:15:50 ... expected string = "test"
15:15:50 
15:15:51 
15:15:51 ----------------------------------------------------------------------
15:15:51 FAIL: docker_cli_run_test.go:4309: DockerSuite.TestRunNamedVolumesFromNotRemoved
15:15:51 
15:15:51 docker_cli_run_test.go:4322:
15:15:51     c.Assert(strings.TrimSpace(out), checker.Equals, "test")
15:15:51 ... obtained string = "Template parsing error: template: :1:2: executing \"\" at <.ID>: ID is not a field of struct type *formatter.volumeContext"
15:15:51 ... expected string = "test"
15:15:51 
```

**Comment by thaJeztah** [comment]
Oh! No raw fallback, perhaps? `ID` -> `Id`

**Comment by vdemeester** [comment]
@thaJeztah the volume type I use (`types.Volume`) have no idea (and there is no idea in `docker volume ls`), I did miss that :stuck_out_tongue_closed_eyes:

**Comment by thaJeztah** [comment]
Think it makes sense to have this here as well, so moved to code review.

Already played with it a bit, and all looks to work properly (with some limited testing)

File: api/client/formatter/network.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	c.addHeader(nameHeader)
+	return c.n.Name
+}
+
+func (c *networkContext) Driver() string {
+	c.addHeader(driverHeader)
+	return c.n.Driver
+}
+
+// YES/NO vs true/false vs 1/0 ?
```
Comment (vdemeester) [inline api/client/formatter/network.go]: @thaJeztah I wasn't sure what to put here for boolean values. How do we usually print them ?

Comment (thaJeztah) [inline api/client/formatter/network.go]: Hm, good question, should have to check in other format outputs

**Comment by vdemeester** [comment]
Updated with integration tests :angel: 
/cc @runcom @cpuguy83 @LK4D4 for code reviews :angel:

File: api/client/cli.go [hunk 2]
```diff
@@ -123,6 +123,18 @@ func (cli *DockerCli) ImagesFormat() string {
 	return cli.configFile.ImagesFormat
 }
 
+// NetworksFormat returns the format string specified in the configuration.
+// String contains columns and format specification, for example {{ID}}\t{{Name}}
+func (cli *DockerCli) NetworksFormat() string {
+	return cli.configFile.NetworksFormat
+}
```
Comment (dnephin) [inline api/client/cli.go]: Do we need to expose all of these directly? I was really surprised to see an `ImageFormat()` method on `DockerCli`.

You can access this already from `dockerCli.ConfigFile().NetworksFormat`

Comment (vdemeester) [inline api/client/cli.go]: Hum true :sweat_smile: I just mimic the current behavior but I could rework them all to use `dockerCli.ConfigFile().NetworksFormat` I think :wink:

**Comment by vdemeester** [comment]
Rebased and updated 👼

**Comment by dnephin** [comment]
LGTM

**Comment by dnephin** [comment]
retest this please

**Comment by thaJeztah** [comment]
LGTM

**Comment by thaJeztah** [comment]
@vdemeester can you add the docs changes to this PR?

**Comment by vdemeester** [comment]
@thaJeztah on it 👼

**Comment by thaJeztah** [comment]
Thanks! docs look good; can you update the man pages as well?

**Comment by vdemeester** [comment]
@thaJeztah damn I forgot to update those :sweat_smile:

**Comment by thaJeztah** [comment]
LGTM!

**Comment by vdemeester** [comment]
Author LGTM (for docs :angel:)

**Pull request merged**